### ISSUE TITLE
Error launching application on iPhone 12 Pro Max

### ISSUE THREAD

**Issue by HassanChmsdn** [flutter/flutter#69632]
**After updating xcode to 12.1 my flutter app was not able to install pods and launch on iPhone 12 pro max or earlier:**
I tried to change the IOS Deployment Target to higher than 8.0 but same error!

I tried these steps to reproduce it but was not successful:

```
rm ios/Podfile
flutter clean
flutter run

and also 
https://stackoverflow.com/a/63283863/4922664

```

flutter run --verbose:
```
/Library/Ruby/Gems/2.6.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'

/Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.0/lib/cocoapods/command.rb:52:in `run'
                        /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.0/bin/pod:55:in `<top
(required)>'
                        /usr/local/bin/pod:23:in `load'
                        /usr/local/bin/pod:23:in `<main>'
                    
[ +237 ms] Error output from CocoaPods:
                    ↳
[        ] 
               [!] Automatically assigning platform `iOS` with version `10.0` on target
`Runner`
               because no platform was specified. Please specify a platform for this
target in your
               Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
           
[   +4 ms] Error running pod install
[   +2 ms] Error launching application on iPhone 12 Pro Max.
[   +5 ms] "flutter run" took 24,592ms.
[   +9 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      RunCommand.runCommand
(package:flutter_tools/src/commands/run.dart:591:7)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand
           (package:flutter_tools/src/runner/flutter_command.dart:860:18)
           #3      _rootRunUnary (dart:async/zone.dart:1198:47)
           #4      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #5      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #6      Future._propagateToListeners.handleValueCallback
           (dart:async/future_impl.dart:696:45)
           #7      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #8      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #11     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
           #12     _rootRunUnary (dart:async/zone.dart:1198:47)
           #13     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #14     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #15     Future._propagateToListeners.handleValueCallback
           (dart:async/future_impl.dart:696:45)
           #16     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #17     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #18     Future._asyncCompleteWithValue.<anonymous closure>
           (dart:async/future_impl.dart:567:7)
           #19     _rootRun (dart:async/zone.dart:1190:13)
           #20     _CustomZone.run (dart:async/zone.dart:1093:19)
           #21     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #22     _CustomZone.bindCallbackGuarded.<anonymous closure>
(dart:async/zone.dart:1037:23)
           #23     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #24     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #25     _runPendingImmediateCallback
(dart:isolate-patch/isolate_patch.dart:118:13)
           #26     _RawReceivePortImpl._handleMessage
(dart:isolate-patch/isolate_patch.dart:169:5)
           
           
[ +120 ms] ensureAnalyticsSent: 112ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[   +8 ms] Shutdown hooks complete
[        ] exiting with code 1
```

flutter doctor -v
```
[✓] Flutter (Channel stable, 1.20.4, on Mac OS X 10.15.7 19H2, locale en-GB)
    • Flutter version 1.20.4 at /Users/hassanchamseddine/Documents/github/flutter
    • Framework revision fba99f6cf9 (7 weeks ago), 2020-09-14 15:32:52 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/hassanchamseddine/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    • CocoaPods version 1.10.0

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (1 available)
    • iPhone 12 Pro Max (mobile) • 13029BE9-4FFD-4F4F-B983-9C2B9795CADD • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-14-1 (simulator)

• No issues found!
```

</details>

**I have tried to upgrade flutter to 1.22.3 but the app crashes for some incompatible plugins which was not clear how to reproduce it as well. so i downgraded it again to 1.20.4, it would be nice if there is a way to update it to latest flutter version here is my pubspec.yaml**

```

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
#
#
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+8

environment:
  sdk: ">=2.2.2 <3.0.0"

dependencies:
  logs:
    git: https://github.com/pq/logs
  flutter:
    sdk: flutter
  dio: ^3.0.10
  http: 0.12.0+4
  html_unescape: 1.0.1+3
  provider: 4.0.4
  after_layout: 1.0.7+2
  font_awesome_flutter: 8.5.0
  sentry: ">=3.0.0 <4.0.0"
  path: 1.7.0
  vector_math: 2.0.8
  carousel_pro: 1.0.0
  flutter_widget_from_html_core: 0.3.2+1
  configurable_expansion_tile: 1.0.0
  timeago: 2.0.26
  share: 0.6.4+2
  validate: 1.7.0
  country_pickers: 1.1.0
  shared_preferences: 0.5.7+2
  firebase_messaging: 6.0.9
  firebase_analytics: 5.0.11
  transparent_image: 1.0.0
  pull_to_refresh: 1.5.8
  localstorage: 3.0.2+5
  adobe_xd: 1.0.0+1
  notification_permissions: 0.4.4
  flare_splash_screen: 3.0.1
  rate_my_app: 0.6.1+2
  flutter_range_slider: 1.4.0
  page_indicator: 0.3.0
  extended_image: 0.8.0
  flutter_screenutil: 1.0.2
  fluttertoast: ^4.0.1
  intro_slider: ^2.3.1
  url_launcher: ^5.4.7
  firebase_core: 0.4.4
  firebase_auth: 0.15.4
  firestore_ui: 1.7.2
  flutter_webview_plugin: 0.3.10+1
  connectivity: 0.4.6+2
  random_string: 2.0.1
  google_maps_flutter: 0.5.23+1
  location: 3.0.2
  geolocator: 6.0.0+4
  geocoder: 0.2.1
  cached_network_image: 2.0.0
  image_picker: 0.6.3+1
  fab_circular_menu: 0.1.4
  uuid: 2.0.4
  uni_links: 0.2.0
  flutter_staggered_grid_view: 0.3.0
  device_apps: 1.0.9
  auto_size_text: 2.1.0
  lazy_load_scrollview: 1.0.1
  flutter_swiper: 1.1.6
  flutter_widgets: 0.1.11
  video_player: 0.10.7
  flutter_spinkit: 4.1.1+1
  percent_indicator: 2.1.1+1
  #youtube_player_flutter: ^5.0.0+1
  animated_text_kit: 2.0.0
  wave: 0.0.8
  http_auth: 0.2.9
  firebase_admob: ^0.9.0+10
  facebook_audience_network: ^0.4.0
  international_phone_input: 1.0.2
  pin_code_fields: 2.3.0+3
  cupertino_icons: 0.1.3
  flutter_facebook_login: 3.0.0
  package_info: 0.4.1
  webview_flutter: 0.3.22
  rubber: 0.4.0
  flutter_localizations:
    sdk: flutter
  carousel_slider: 1.4.1
  custom_splash: 0.0.2
  apple_sign_in: 0.1.0
  google_sign_in: 4.1.1
  google_fonts: 0.3.9
  onesignal_flutter: 2.4.1
  flutter_statusbarcolor: 0.2.3
  app_settings: 3.0.0+1
  razorpay_flutter: 1.2.2
  shimmer: 1.0.1
  animated_splash: 1.0.0
  firebase_database: 3.1.1
  event_bus: 1.1.1
  launch_review: 2.0.0
  country_code_picker: 1.2.4
  edge_alert: 0.0.1


dev_dependencies:
  flutter_launcher_icons: 0.7.4
  flutter_test:
    sdk: flutter
  pedantic: 1.8.0+1 # Analysis


# Run this script to generate the app icon: flutter pub pub run flutter_launcher_icons:main
flutter_icons:
  android: "launcher_icon"
  ios: true
  image_path: "assets/images/app_icon.png"

# For information on the generic Dart part of this file, see the
# following page: https://www.dartlang.org/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:
  uses-material-design: true
  assets:
    - lib/common/
    - assets/icons/categories/
    - assets/icons/tabs/
    - assets/icons/profile/
    - assets/icons/payment/
    - assets/images/
    - assets/images/country/
    - assets/images/luxury/
  fonts:
    - family: Raleway
      fonts:
        - asset: assets/fonts/Raleway-Regular.ttf
        - asset: assets/fonts/Raleway-Medium.ttf
        - asset: assets/fonts/Raleway-Bold.ttf
        - asset: assets/fonts/Raleway-Light.ttf
        - asset: assets/fonts/Raleway-Thin.ttf
    - family: Nexa
      fonts:
        - asset: assets/fonts/Nexa-Bold.otf
        - asset: assets/fonts/Nexa-Light.otf
    - family: Roboto
      fonts:
        - asset: assets/fonts/Roboto-Regular.ttf
        - asset: assets/fonts/Roboto-Medium.ttf
        - asset: assets/fonts/Roboto-Bold.ttf
        - asset: assets/fonts/Roboto-Light.ttf
        - asset: assets/fonts/Roboto-Thin.ttf
    - family: Mirantie
      fonts:
        - asset: assets/fonts/mirantie.ttf

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.io/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.io/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
        
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.io/custom-fonts/#from-packages
flutter_intl:
  enabled: true

```

**Comment by HassanChmsdn** [comment]
I  have fixed this issue by running:  

```
flutter clean
pod cache clean  --all
rm -rf ios/Flutter/Flutter.framework

flutter pub get
pod install
flutter run```

thanks to [#20045](https://github.com/flutter/flutter/issues/20045#issuecomment-685282521)

**Comment by darshankawar** [comment]
@HassanChmsdn Based on your last comment, I am going ahead and closing this as resolved.
If you disagree, write in comments and I'll reopen.
Thanks.

**Comment by virenvvarasadiya** [comment]
@HassanChmsdn solution does not work for me. i am just getting single line error and "Error launching application on Viren’s iPhone." and then application is opening in device but it is then disconnect with vscode. @darshankawar

**Comment by andylai** [comment]
same issue here, any solution?

**Comment by ductranit** [comment]
I got the same issue with the latest `master` version. When I run debug from vscode, it shows error `Error launching application on iPhone` and stops connect in vscode. 
But weird that it works on another device (my ipad)
My `flutter doctor`
```
[✓] Flutter (Channel master, 1.26.0-18.0.pre.126, on Mac OS X 10.15.7 19H114 darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.52.1)
[✓] Connected device (3 available)

• No issues found!
```

**Comment by akarabach** [comment]
the same issue on my device

**Comment by talatkuyuk** [comment]
Error output from CocoaPods:
↳
    [!] Automatically assigning platform `iOS` with version `9.0` on target `Runner` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
Error running pod install
Error launching application on iPhone 12 Pro Max.
Exited (sigterm)

This is the message I have received. I have a problem with that issue.

**Comment by akarabach** [comment]
Yup, I had the same issue with signing in the app, but Ive found out about it after a couple of hours on stackoverflow! I've run xcode and sign in the app again, for sure we need more intuitive error instead of this one!

**Comment by lamadipen** [comment]
For me the issue was solved after I uncomment the `platform :ios, '9.0'` in Podfile  under ios directory.

**Comment by christxph** [comment]
For me doing this fixed the problem:

1. Setting platform to 10.0 inside the Podfile (`platform :ios, '10.0' `)
2. Doing a combination of `flutter clean`, `pod cache clean  --all` and `pod install`
3. Opening the project inside XCode

Only after doing the last step, `flutter run` was working fine again.

**Comment by emiliodallatorre** [comment]
> I tried all of these fixes but I still get this error. Any update?

Probably, this is a duplicate of #72161. You should check with `flutter run --verbose` if your app stops at `Application launched on the device. Waiting for observatory port.`.

**Comment by BringTheChill** [comment]
> > I tried all of these fixes but I still get this error. Any update?
> 
> Probably, this is a duplicate of #72161. You should check with `flutter run --verbose` if your app stops at `Application launched on the device. Waiting for observatory port.`.

yes, it's stops there with an error but apps is working (even if it's not running on IDE). what whould I do?

**Comment by ablbol** [comment]
I tried everything mentioned here but noting worked for me!!

**Comment by emiliodallatorre** [comment]
> > > I tried all of these fixes but I still get this error. Any update?
> > 
> > 
> > Probably, this is a duplicate of #72161. You should check with `flutter run --verbose` if your app stops at `Application launched on the device. Waiting for observatory port.`.
> 
> yes, it's stops there with an error but apps is working (even if it's not running on IDE). what whould I do?

I wish I could find an answer...

**Comment by DJSL1NKY** [comment]
I have the same issue with my iPhone 12 Pro Max running iOS 14.4.2 and Flutter 2.0.3.

**Comment by BringTheChill** [comment]
@darshankawar even if the issue is closed the problem still exists. can you tell us if someone is working to fix it or something?

**Comment by jmagman** [comment]
If you're seeing this can you please please file a new issue and completely fill out the template?  No one monitors closed issues.  I'd be really interested to look at this problem if you're on Flutter 2, we've fixed a bunch of issues in this area (most of which are available on Flutter 2) and I'd like to track down remaining problems.

**Comment by BringTheChill** [comment]
> If you're seeing this can you please please file a new issue and completely fill out the template? No one monitors closed issues. I'd be really interested to look at this problem if you're on Flutter 2, we've fixed a bunch of issues in this area (most of which are available on Flutter 2) and I'd like to track down remaining problems.

thank you man. check #79987

**Comment by aleksoft1** [comment]
I uncomment the `platform :ios, '9.0' `in Podfile under ios directory and change to '10.0' to fix the issue.

**Comment by fuhrthom** [comment]
I have the same problem with my iPhone 12 Pro. On my old iPhone 7 its working like charm.

**Comment by jmagman** [comment]
@fuhrthom
> If you're seeing this can you please please file a new issue and completely fill out the template? No one monitors closed issues. I'd be really interested to look at this problem if you're on Flutter 2, we've fixed a bunch of issues in this area (most of which are available on Flutter 2) and I'd like to track down remaining problems.

**Comment by scortumee** [comment]
I had the same problem with my iPhone 8 (IOS 14.6) on VScode. All I did was to run the app through Xcode once and the problem was fixed.

**Comment by cliff-mvelase** [comment]
run:
flutter clean 
flutter upgrade

**Comment by fuhrthom** [comment]
> run:
> flutter clean
> flutter upgrade

does fix it, thanks! what was the problem?

**Comment by cliff-mvelase** [comment]
> > run:
> > flutter clean
> > flutter upgrade
> 
> does fix it, thanks! what was the problem?

yes it does. think there was a bug on the previous update. You may also need to run:
       flutter pub upgrade 
on your project.
and  to set up CocoaPods for ARM macOS, run:
       arch -x86_64 sudo gem install ffi

**Comment by nijatmursali** [comment]
In my case, if I do `flutter run` it does not work and gives that error, but if I do `flutter run --release` it works fine. Weird, isn't it?

**Comment by jmagman** [comment]
Please file a new issue and give us enough information to fix it for everyone instead of needing these work arounds!

> If you're seeing this can you please please file a new issue and completely fill out the template? No one monitors closed issues. I'd be really interested to look at this problem if you're on Flutter 2, we've fixed a bunch of issues in this area (most of which are available on Flutter 2) and I'd like to track down remaining problems.
